tests/interpolation/test_custom_resolvers.py

```diff
@@ -8,22 +8,57 @@
 from tests.interpolation import dereference_node
 
 
-def test_register_resolver_twice_error(restore_resolvers: Any) -> None:
+def test_register_resolver_twice_error_new_callable(restore_resolvers: Any) -> None:
     def foo(_: Any) -> int:
         return 10
 
     OmegaConf.register_new_resolver("foo", foo)
-    with raises(AssertionError):
+    with raises(ValueError, match=re.escape("resolver 'foo' is already registered")):
         OmegaConf.register_new_resolver("foo", lambda _: 10)
 
 
+@mark.parametrize("first_use_cache", [True, False])
```

Comment:
since the behavior of resiter_new_resolver is much simpler now, I think this test is no longer needed.

Reply:
Ok, simplified in e5a490b